torch/onnx/symbolic_opset9.py

```diff
@@ -2248,33 +2248,57 @@ def topk(g, self, k, dim, largest, sorted, out=None):
 
 
 def to(g, self, *args):
-    # ONNX doesn't have a concept of a device, so we ignore device casts
-    if len(args) == 4:
-        if args[0].node().kind() == "prim::device" or args[0].type().isSubtypeOf(ListType.ofInts()):
+
+    def is_aten_to_device_only(*args):
+        if len(args) == 4:
             # aten::to(Tensor, Device, bool, bool, memory_format)
-            return self
+            if args[0].node().kind() == "prim::device" or \
+                    args[0].type().isSubtypeOf(ListType.ofInts()) or \
+                    (sym_help._is_value(args[0]) and
+                        args[0].node().kind() == "onnx::Constant" and
+                        isinstance(args[0].node()["value"], str)):
+                return True
+        elif len(args) == 5:
+            # aten::to(Tensor, Device, ScalarType, bool, bool, memory_format)
+            dtype = sym_help._get_const(args[1], "i", "dtype")
+            if dtype is None:
```

Comment:
ditto, remove if

Reply:
As explained in an earlier comment: It's more compact and just as clear. I'd argue that it's even clearer since there's less control flow that the user has to understand as they read the code. With the current code for example, what happens in the `else` case is pretty far away from here and not as obvious as if we just return here in both cases.